torchnlp/metrics/rouge.py

```diff
@@ -0,0 +1,332 @@
+                c[i, j] = c[i - 1, j - 1] + func(k + 1) - func(k)
+                w[i, j] = k + 1
+            else:
+                c[i, j] = max(c[i - 1, j], c[i, j - 1])
+                w[i, j] = 0
+    return c[n, m]
+
+
+def get_rouge_n(evaluated_sentences, reference_sentences, n=2, exclusive=True):
+    """
+    Computes ROUGE-N of two text collections of sentences, namely evaluated_sentences and
+        reference senteces
+    Args:
+        evaluated_sentences: The sentences that have been produced by the
+                           summarizer
+        reference_sentences: The sentences from the referene set
+        n: Size of ngram.  Defaults to 2.
+    Returns:
+        tuple. (f1, precision, recall) for ROUGE-N
+    """
+    if len(evaluated_sentences) <= 0:
+        raise ValueError("Hypothesis set is empty.")
+    if len(reference_sentences) <= 0:
+        raise ValueError("reference set is empty.")
+    for eval_sentence, ref_sentence in zip(evaluated_sentences, reference_sentences):
+        eval_ngrams = _get_ngrams(n, eval_sentence, exclusive)
+        ref_grams = _get_ngrams(n, ref_sentence, exclusive)
+        ref_count = len(ref_grams)
+        eval_count = len(eval_ngrams)
+        overlapping_ngrams = eval_ngrams.intersection(ref_grams)
+        overlapping_count = len(overlapping_ngrams)
+    return _frp_rouge_n(eval_count, ref_count, overlapping_count)
+
+
+def get_rouge_l_summary_level(evaluated_sentences, reference_sentences):
+    """
+    Computes ROUGE-L of two summary-level sentences, namely
+        evaluated_sentences and reference senteces
+    Reference: https://www.aclweb.org/anthology/W04-1013.pdf
+    Args:
+        evaluated_sentences: List of sentences that have been produced by the summarizer
+        reference_sentences: List of sentences from the reference set
+    Returns:
+        dictionary. {'f': f1_score, 'p': precision, 'r': recall} for ROUGE-L
+    """
+    if len(evaluated_sentences) <= 0:
+        raise ValueError("Hypothesis set is empty.")
+    if len(reference_sentences) <= 0:
+        raise ValueError("reference set is empty")
+
+    llcs = 0
+    ref_words = [w for s in reference_sentences for w in s.split()]
+    eval_words = [w for s in evaluated_sentences for w in s.split()]
+    m = len(set(ref_words))
+    n = len(set(eval_words))
+
+    reference_sentences = [s.split() for s in reference_sentences]
+    evaluated_sentences = [s.split() for s in evaluated_sentences]
+    for evaluated_sentence in evaluated_sentences:
+        union_lcs = set()
+        for reference_sentence in reference_sentences:
+            union_lcs |= set(lcs_seq(reference_sentence, evaluated_sentence))
+        llcs += len(union_lcs)
+    return _frp_rouge_l(llcs, m, n)
+
+
+def get_rouge_w(evaluated_sentence, reference_sentence,
+                f=lambda x: x**2, inv_f=lambda x: math.sqrt(x)):
+    """
+    Computes ROUGE-W of two sequences, namely evaluated_sentence and reference sentece
```

Comment:
Can you please try to build the documentation via something similar to `cd docs; make html`? You'll need to double-check this can be parsed for ReadTheDocs.